crates/bootstrap_srv/src/bin/kitsune2-bootstrap-srv.rs

```diff
@@ -109,6 +109,12 @@ fn main() {
         Config::testing()
     };
 
+    if args.tls_cert.is_some() || args.tls_key.is_some() {
```

Comment:
Turns out we do need this. I have no desire to keep endlessly fiddling with which features are enabled and it's super fragile because we won't detect this on CI and it's a runtime decision which provider to pick. With multiple providers available, it defaults to picking none. Probably sensible but then let's just decide for it